### PR TITLE
Feat sign up page

### PR DESCRIPTION
**Pull request by yuanliuddd**
Add : general css file :  all variable (colors, font-size); 
Finish : Step 1 pages

### CODE REVIEW

**Review by sebastienng — CHANGES_REQUESTED** [review]
Logo already imported

setSteps not handled in the stepOnIdentifier.jsx

File: client/src/pages/Signup.jsx [hunk 1]
```diff
@@ -4,46 +4,68 @@ import { useNavigate } from "react-router-dom";
...
+    jobs: {
+      category: "",
+      title: "",
+      skills: [],
+      experience: 0,
+    },
+    language: [],
+    description: "",
+    location: "",
+  };
```
Comment (kevbarns) [inline client/src/pages/Signup.jsx]: Signup data should only have the required data from the page, i.e the 4 first parameters.

File: client/src/pages/Signup.jsx [hunk 2]
```diff
@@ -4,46 +4,68 @@ import { useNavigate } from "react-router-dom";
...
   return (
     <div>
-      <h1>Sign Up</h1>
-      <form onSubmit={handleFormSubmission} className="auth__form">
+      {steps === 1 ? <StepOneIdentifier setSignUpData={setSignUpData} /> : null}
+
+      {/* <form onSubmit={handleFormSubmission} className="auth__form">
         <label htmlFor="input-username">Username</label>
         <input
           id="input-username"
```
Comment (kevbarns) [inline client/src/pages/Signup.jsx]: I really encourage you to have a look at the `formik` form library : it's easy to setup and provide handlers & validation out of the box (validation schema can be made using `Yup` package)

Comment (kevbarns) [reply client/src/pages/Signup.jsx]: example : https://github.com/mission-apprentissage/matcha/blob/master/ui/src/pages/Authentification/Connexion.jsx

File: client/src/components/SignInSteps/StepTwo.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              promotions.
+            </label>
+            <button type="submit"> Sign up </button>
+          </form>
+        )}
+      </Formik>
+      <p>
+        By signing up you agree with our <span>Terms of Use</span> and{" "}
+        <span>Privacy Policy</span> .
+      </p>
```
Comment (kevbarns) [inline client/src/components/SignInSteps/StepTwo.jsx]: This part is supposed to be right bellow the password field.

File: client/src/config/signUpValidationSchema.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    .required("Required")
+    .max(15, "Must be 15 characters or less"),
+  lastName: Yup.string()
+    .min(2, "Must be at least 2 characters")
+    .required("Required")
+    .max(20, "Must be 20 characters or less"),
+  email: Yup.string()
+    .email("Invalid email address")
+    .required("Required")
+    .matches(/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g),
```
Comment (kevbarns) [inline client/src/config/signUpValidationSchema.js]: Matches is not required here as the email() method is already enough.

**Comment by kevbarns** [comment]
@yuanliuddd @sebastienng  place all related code for steps 2 in another branch and remove them from this branch

File: client/src/config/signUpValidationSchema.js (hunk 4 above)
Comment (yuanliuddd) [reply client/src/config/signUpValidationSchema.js]: ok

Comment (yuanliuddd) [reply client/src/config/signUpValidationSchema.js]: fixed done

File: client/src/components/SignInSteps/StepTwo.jsx (hunk 3 above)
Comment (yuanliuddd) [reply client/src/components/SignInSteps/StepTwo.jsx]: yes , but i put  **agreement (input checkbox)**  in the form along with other inputs  , so this part have to be outside of the form . So the position isn't right . I'm planning to use position relative / absolute  to correct it

**Review by sebastienng — APPROVED** [review]

**Comment by kevbarns** [comment]
Faut merger la PR la, y'a trop de code.

**Pull request merged**